docs/messaging/gcp-pubsub.md

```diff
@@ -16,7 +16,15 @@ For Google Cloud Pub/Sub, the following additional attributes are defined:
 <!-- semconv messaging.gcp_pubsub(full,tag=tech-specific-gcp-pubsub) -->
 | Attribute  | Type | Description  | Examples  | Requirement Level |
 |---|---|---|---|---|
+| `messaging.gcp_pubsub.message.ack_result` | string | The subscriber's acknowledgement of a message. If missing, the subscriber did not chose whether or not to acknowledge the message. | `ack`; `nack` | Conditionally Required: If the subscriber ACKs or NACKs the message. |
 | [`messaging.gcp_pubsub.message.ordering_key`](../attributes-registry/messaging.md) | string | The ordering key for a given message. If the attribute is not present, the message does not have an ordering key. | `ordering_key` | Conditionally Required: If the message type has an ordering key set. |
+
+`messaging.gcp_pubsub.message.ack_result` MUST be one of the following:
```

Comment:
In a different PR, @alevenberg proposes to encode this information in the operation name: https://github.com/open-telemetry/semantic-conventions/pull/634/files#diff-b4525d57b2e14f26eee734f14cac1c86a3d3a5f8e0b0fdca24c75c585402661fR25

We should settle on one of those two approaches, to avoid duplicating data.

Reply:
Span name is always a duplication. 
we should have an attribute as well - we don't want users to parse status from the operation name to query nack-ed messages. Also, we need an attribute for metrics.

Reply:
I'm not referring to the span name, just to attributes. @alevenberg proposes to have GCP specific operation names `messaging.operation=subscribe|modack|ack|nack`, whereas this PR proposes `messaging.gcp_pubsub.message.ack_result=ack|nack`.

Maybe it makes sense to have this duplication, but I'd just like GCP Pub/Sub experts to confirm.

Reply:
Thanks for the feedback! I should have checked the open PRs - I like that PR's approach, I'll close this one in favour of that.

Reply:
Sorry I missed this PR! 

I actually wanted to add a similar attribute for the streaming pull `messaging.gcp_pubsub.message.result` where the values are ack/nack/expired.

That's awesome that you implemented the Otel tracing for the Python/Java clients.

I can try and put together a streaming pull PR so you can see what the spec is for the C++/Go/Node clients.

Reply:
thanks for the clarification, @pyohannes !

@alevenberg I agree that settle is too vague, but both ack and nack sound like types of settlement.

What I was going to do for azure is
- `messaging.operation = settle`
- `span name = <destination> complete | defer | ...`
- `messaging.servicebus.disposition_status = complete`

This way backends that don't know about servicebus specifics can still know what's the generic operation being performed, but also users can get everything they need to know.

If we go `ack`, `nack` as `messaging.operation` for PubSub, we should do the same for RabbitMQ and then what we have for `settle` operation in the spec now falls apart.

Reply:
@alevenberg a spec for the streaming pull would be great! I hope I haven't stepped on anyone's toes with the python/java instrumentation. [Here](https://github.com/open-telemetry/opentelemetry-java-instrumentation/pull/10546) is my instrumentation for the java client - but would you prefer it to be part of the actual [client library](https://github.com/googleapis/java-pubsub) instead? Is someone else already working on that, or should I try to port my stuff over there?

Reply:
Hi @samwright, I'm helping with the tracing efforts for Pub/Sub. We do have plans to add native instrumentation into the Java client library (as you linked above). My recommendation would be to hold off on your PR since there's likely more information that can be traced internally. Also, we've standardized on an augmentation of the W3C trace context implementation with Pub/Sub message attributes, and that currently wouldn't be compatible with the PR you have out.

Reply:
Hi @hongalex, that's great to hear that the client will natively support otel! I don't suppose you know vaguely when the python and java clients will be updated?

In the meantime I'll close my PR.

Reply:
Sorry just saw this. We're looking to start work on this soon, but no ETA to share yet. We've been working on finalizing our tracing design to align with [messaging semantic conventions](https://opentelemetry.io/docs/specs/semconv/messaging/messaging-spans/) which is still under development but fairly standardized now.